src/group_theory/p_group.lean

```diff
@@ -14,71 +16,143 @@ then the number of fixed points of the action is congruent mod `p` to the cardin
 It also contains proofs of some corollaries of this lemma about existence of fixed points.
 -/
 
+open_locale big_operators
+
+variables (p : ℕ) (G : Type*) [group G]
+
+/-- A p-group is a group in which every element has prime power order -/
+def is_p_group : Prop := ∀ g : G, ∃ k : ℕ, g ^ (p ^ k) = 1
```

Comment:
Sure, although making `is_p_group` a class seems to make some things in `sylow.lean` more painful. For example, if you assume `fintype.card H = p ^ n` (rather than `is_p_group H`, because `n` is mentioned elsewhere in the statement), then might need to make an `is_p_group H` instance, which could be tricky in some of the more term-mode-y proofs.

Reply:
Some other thoughts:
- Treating `is_p_group` as an instance means that the `.` notation can't be used.
- One advantage of making `is_p_group` an instance is that there is one less explicit parameter you have to pass in. But from my refactoring, it seems like you now have to pass in `p` and maybe also `G`, so you don't really gain anything in this regard.
- The main advantages I see are the subgroup and quotient instances. But I'm not sure if those outweigh the stuff that I've mentioned.